### ISSUE TITLE
M2.1.1 : Rest API V1 Error when saving shipping address with no country is not clear.

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#6592]
When you attempt to save shipping information without a country ID, you get the message `Shipping address is not set`. You can see this in `public/vendor/magento/module-checkout/Model/ShippingInformationManagement.php:142`:

```
if (!$address->getCountryId()) {
    throw new StateException(__('Shipping address is not set'));
}
```

A more logical error message would be `Country ID not set`. It would have saved me at least an hour of debugging ;-)

**Comment by veloraven** [comment]
@kanduvisla thank you for your feedback.
Please use our issue reporting template to describe this issue. The template specifies the version and configuration information we need to assess your issue. See the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) for more information.

**Comment by veloraven** [comment]
Internal ticket MAGETWO-61669

**Comment by magento-engcom-team** [comment]
@kanduvisla, thank you for your report.
The issue is already fixed in 2.2.0
